syft/source/source.go

```diff
@@ -304,6 +306,50 @@ func NewFromImage(img *image.Image, userImageStr string) (Source, error) {
 	}, nil
 }
 
+func (s *Source) ID() string {
+	return s.id
+}
+
+func (s *Source) SetID() {
+	if s.Metadata.Scheme != ImageScheme {
+		// How do we generate ID for non-image sources?
+		s.id = digest.FromString(s.Metadata.Path).String()
+		return
+	}
+
+	// calcuate chain ID for image sources
+	// https://github.com/opencontainers/image-spec/blob/main/config.md#layer-chainid
+	s.id = calculateChainID(s.Image)
+
+	if s.id == "" {
+		// TODO what happens here if image has no layers?
+		s.id = digest.FromString(s.Metadata.ImageMetadata.UserInput).String()
+	}
+	return
+}
+
+func calculateChainID(img *image.Image) string {
```

Comment:
hmm, trying to understand a bit more,.. have to preface this with i don't think i've fully grasped the proposal.. 
i think the hash of the container (well the one is used to download the container and identify it), isn't the layer chain ID? - but the sha256 of the OCI manifest

Reply:
Yea good point - probably will use this as the fallback if syft can determine the image was not pulled and is locally built. Thanks for the comment!